### PR TITLE
feat: update artifact manager for qct .net to include private package support

### PR DESCRIPTION
**Pull request by amanprak**
## Problem
Update artifact manager for qct .net to include private package support

## Solution

ArtifactManager Updates:

- store core compatible assembly files in thirdpartypackages folder under references
- update the requirement.json created with relative path to files in the above folder.

## Testing
Testing positive case, verify artifact.zip was created properly

example artifact.zip after change.
[artifact.zip](https://github.com/user-attachments/files/19558786/artifact.zip)

Screenshot 2025-04-01 at 4 00 04 PM

Screenshot 2025-04-01 at 4 00 22 PM

Screenshot 2025-04-01 at 4 00 33 PM

Also test negative case where PackageReferences not provided in the request, make sure requirement.json and artifact content still created properly.

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by mrkdeng — APPROVED** [review]

File: server/aws-lsp-codewhisperer/src/language-server/netTransform/artifactManager.ts [hunk 1]
```diff
@@ -20,26 +21,52 @@ export class ArtifactManager {
         this.logging = logging
         this.workspacePath = workspacePath
     }
+
     async createZip(request: StartTransformRequest): Promise<string> {
+        this.logging.log('Starting createZip process...')
```
Comment (pranav-firake) [inline server/aws-lsp-codewhisperer/src/language-server/netTransform/artifactManager.ts:26]: are these logs from debugging or do we want these to surface to customers?

- Earlier we had these level of logs but we had to delete these because we were logging too much details that customer would not need to see.

File: server/aws-lsp-codewhisperer/src/language-server/netTransform/artifactManager.ts [hunk 2]
```diff
@@ -20,26 +21,52 @@ export class ArtifactManager {
...
         await this.copySolutionConfigFiles(request)
+
+        this.logging.log('Removing duplicate NuGet packages folder...')
         await this.removeDuplicateNugetPackagesFolder(request)
-        return await this.zipArtifact()
+
+        this.logging.log('Creating final zip file...')
+        const zipPath = await this.zipArtifact()
+
+        this.logging.log(`Zip creation completed. Path: ${zipPath}`)
```
Comment (pranav-firake) [inline server/aws-lsp-codewhisperer/src/language-server/netTransform/artifactManager.ts:40]: we are already logging zipPath at [link](https://github.com/aws/language-servers/blob/main/server/aws-lsp-codewhisperer/src/language-server/netTransform/artifactManager.ts#L144) , this will be duplicate logs in customer facing logs

**Comment by pranav-firake** [comment]
just confirming if we want to merge this PR to main branch. should we target to transform-ga branch?

File: server/aws-lsp-codewhisperer/src/language-server/netTransform/artifactManager.ts (hunk 1 above)
Comment (cslong) [reply server/aws-lsp-codewhisperer/src/language-server/netTransform/artifactManager.ts:26]: I'll remove the extra logging.

**Review by pranav-firake — APPROVED** [review]

File: server/aws-lsp-codewhisperer/src/language-server/netTransform/artifactManager.ts (hunk 2 above)
Comment (cslong) [reply server/aws-lsp-codewhisperer/src/language-server/netTransform/artifactManager.ts]: removed the logging

File: server/aws-lsp-codewhisperer/src/language-server/netTransform/artifactManager.ts [hunk 3]
```diff
@@ -133,6 +164,39 @@ export class ArtifactManager {
         } as RequirementJson
     }
 
+    private processPrivatePackages(
```
[the remark this answers is not in the document]
Comment (cslong) [reply server/aws-lsp-codewhisperer/src/language-server/netTransform/artifactManager.ts]: added tests

**Review by pranav-firake — APPROVED** [review]

**Review by francescoopiccoli — APPROVED** [review]
